src/shared_utilities/models.jl

```diff
@@ -447,6 +447,26 @@ function initialize(model::AbstractModel{FT}) where {FT}
     return Y, p, coords
 end
 
+
+"""
+    initialize_from_checkpoint(restart_file; model::AbstractModel)
+
+Creates the prognostic and auxiliary states structures, but with unset
+values; constructs and returns the coordinates for the `model` domain.
+We may need to consider this default more as we add diverse components and
+`Simulations`.
+
+TODO: Combine this function with initialize. We don't really need two.
+"""
+function initialize_from_checkpoint(restart_file; model)
```

Comment:
Does this make sense instead?
`initialize(model) = does what it does now`
`initialize(model, checkpoint) = reads from checkpoint, if model hash saved, also checks that

Reply:
oh! One other idea is to think of the checkpoint not as an initialize step, but as a way to set initial conditions. This would be more consistent with what `initialize` does currently (i.e. just makes the structures Y,p). 

In that case, we call initialize like usual, but the checkpoint is handled as a way to set initial conditions. e.g.
```
Y, p, cds = initialize(model)
initial_conditions_from_checkpoint!(Y,restart_file; model)
t0 = initial_time_from_checkpoint(restart_file; model)
```
and then we call set_initial_cache! like usual

Reply:
^^this might be better, because this PR currently uses `initialize` in two different ways (allocating space vs doing that AND setting IC)